src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1390,124 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
   {
     rRuntimeAnnotation_attribute(parsed_class.annotations);
   }
+  else if(attribute_name == "BootstrapMethods")
+  {
+    INVARIANT(
+      !parsed_class.read_attribute_bootstrapmethods,
+      "only one BootstrapMethods argument is allowed in a class file");
+
+    // mark as read in parsed class
+    parsed_class.read_attribute_bootstrapmethods = true;
+    u2 num_bootstrap_methods = read_u2();
+    for(size_t i = 0; i < num_bootstrap_methods; i++)
+    {
+      u2 bootstrap_methodhandle_ref = read_u2();
+      const pool_entryt &entry = pool_entry(bootstrap_methodhandle_ref);
+      u2 num_bootstrap_arguments = read_u2();
+
+      lambda_method_handlet handle;
+      status() << "INFO: parse BootstrapMethod handle "
+               << num_bootstrap_arguments << " #args"
+               << eom;
+      parse_methodhandle(entry, handle);
+      if(
+        handle.handle_type ==
+          method_handle_typet::BOOTSTRAP_METHOD_HANDLE_ALT ||
+        handle.handle_type == method_handle_typet::BOOTSTRAP_METHOD_HANDLE)
+      {
+        // try parsing bootstrap method handle
+        if(num_bootstrap_arguments >= 3)
+        {
+          // each entry contains a MethodHandle structure
+          // u2 tag
+          // u2 reference kind which must be in the range from 1 to 9
+          // u2 reference index into the constant pool
+          //
+          // reference kinds use the following
+          // 1 to 4 must point to a CONSTANT_Fieldref structure
+          // 5 or 8 must point to a CONSTANT_Methodref structure
+          // 6 or 7 must point to a CONSTANT_Methodref or
+          // CONSTANT_InterfaceMethodref structure, if the class file version
+          // number is 52.0 or above, to a CONSTANT_Methodref only in the case
+          // of less than 52.0
+          // 9 must point to a CONSTANT_InterfaceMethodref structure
+
+          // the index must point to a CONSTANT_String
+          //                           CONSTANT_Class
+          //                           CONSTANT_Integer
+          //                           CONSTANT_Long
+          //                           CONSTANT_Float
+          //                           CONSTANT_Double
+          //                           CONSTANT_MethodHandle
+          //                           CONSTANT_MethodType
+          u2 arg_index1 = read_u2();
+          u2 arg_index2 = read_u2();
+          u2 arg_index3 = read_u2();
+
+          // skip rest
+          for(size_t i = 3; i < num_bootstrap_arguments; i++)
+            read_u2();
+
+          const pool_entryt &arg1 = pool_entry(arg_index1);
+          const pool_entryt &arg2 = pool_entry(arg_index2);
+          const pool_entryt &arg3 = pool_entry(arg_index3);
+
+          if(!(arg1.tag == CONSTANT_MethodType &&
+               arg2.tag == CONSTANT_MethodHandle &&
```

Comment:
Why do we check the tag of arg2 and throw the value away instead of storing for later use?

Reply:
this is just to validate that the format is what we expect a lambda entry to look like